sensord/main.cpp

```diff
@@ -48,27 +49,55 @@
 #include "calibrationhandler.h"
 #include "parser.h"
 
-static QtMsgType logLevel = QtWarningMsg;
+static const QtMsgType levelOrder[] = { QtCriticalMsg, QtWarningMsg, QtInfoMsg, QtDebugMsg };
+static const size_t levelCount = sizeof levelOrder / sizeof *levelOrder;
+
+static int logLevel = 3;
+static QtMessageHandler previousMessageHandler;
+
+static int normalizeLevel(QtMsgType type)
+{
+    /* Map QtMsgType enum values to something that hopefully
+     * makes sense in less-than / greater-than sense too. */
+    switch (type) {
+    case QtCriticalMsg:
+        return 0;
+    case QtWarningMsg:
+        return 1;
+    case QtInfoMsg:
+        return 2;
+    default:
+        return 3;
+    }
+}
+
+static void messageOutput(QtMsgType type, const QMessageLogContext &context, const QString &str)
+{
+    if (normalizeLevel(type) <= logLevel)
+        previousMessageHandler(type, context, str);
```

Comment:
This seems a bit haflway version between the message filter and logging categories as it does both, sets the logging levels but then also filter based on logging levels.

What I'd suggest trying is  avoid re-adding qInstallMessageHandler but instead do the same log level adjustment to QLoggingCategory::defaultCategory() in setLogLevel() as the lcSensorFw() has.

Reply:
I'll try that out